### PR TITLE
[WIP] :warning: Server Side Apply Support

### PR DESCRIPTION
**Pull request by DirectXMan12**
This introduces a server-side apply patch type.  Don't use yet, since it requires Kubernetes 1.14.  Breaking since the structure of PatchOptions needs to change.

### CODE REVIEW

**Comment by DirectXMan12** [comment]
Also TODO: switch to specificallyversionedoptions calls

**Comment by DirectXMan12** [comment]
One step in #347

**Comment by DirectXMan12** [comment]
also TODO (eventually): jsonpatch patch type

**Comment by DirectXMan12** [comment]
(this also needs tests before merge)

File: pkg/client/patch.go [hunk 1]
```diff
@@ -72,3 +77,21 @@ func (s *mergeFromPatch) Data(obj runtime.Object) ([]byte, error) {
 func MergeFrom(obj runtime.Object) Patch {
 	return &mergeFromPatch{obj}
 }
+
+// applyPatch uses server-side apply to patch the object.
+type applyPatch struct{}
```
Comment (apelisse) [inline pkg/client/patch.go]: You don't use the `var _ SomeInterface = someImpl{}` pattern? I'm assuming this implements something that I don't know

Comment (DirectXMan12) [reply pkg/client/patch.go]: See the `Patch` interface above

**Comment by apelisse** [comment]
LGTM from my perspective.

**Comment by apelisse** [comment]
Shame on me for forgetting the fieldmanager field ...

**Comment by DirectXMan12** [comment]
for posterity: part of this went in to #435, the rest needs a bit more work.

**Comment by DirectXMan12** [comment]
we've got minimal support currently, but it's a bit clunky.  We'll add a nice wrapper eventually, but closing for now

**Pull request closed without merging**